src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,174 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        //created all variables
+        int count = 0;
+        int secondIndex;
+        int thirdIndex;
+        int fourthIndex;
+        int fifthIndex;
+        int sixthIndex;
+        String nameFromData;
+        String firstNumber;
+        String secondNumber;
+        int[] arr3 = new int[names.length];
+        int day;
+        int month;
+        int year;
+        int day2;
+        int month2;
+        int year2;
+        int first;
+        int second;
+        int salary = 0;
+        int day1;
+        int month1;
+        int day5;
+        int month5;
+        int day6;
+        int month6;
+        int[] arr2 = new int[names.length];
+
+        /*Searching indexes in order to find numbers and names from 'dataFrom',
+        // 'dataTo', 'names[]', 'data[]'
+         text of data from variable 'dataFrom'
+         */
+        fourthIndex = dateFrom.indexOf(" ");
```

Comment:
It seems you are trying to parse dates manually. Instead of using indexes and substring methods, use a DateTimeFormatter to parse dates.